### PR TITLE
Fix README

### PR DESCRIPTION
**Pull request by yuyu2172**

### CODE REVIEW

**Comment by yuyu2172** [comment]
I think these changes can be merged.

File: README.md [hunk 1]
```diff
@@ -63,7 +64,7 @@ img, label = dataset[0]
 
 As found in the example, `random_rotate` is one of the transforms ChainerCV supports. Like other transforms, this is just a
 function that takes an array as input.
-Also, `TransformDataset` is a new dataset class added in ChainerCV that overrides the underlying dataset's `__getitem__` by calling `transform` as post processing.
+Also, `TransformDataset` is a new dataset class added in ChainerCV that overrides the underlying dataset's `__getitem__` by applying `transform` to the values returned by the original `__getitem__`.
```
Comment (Hakuyume) [inline README.md]: `__getitem__` -> :meth:`__getitem__` ?

File: README.md [hunk 2]
```diff
@@ -39,10 +39,11 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
 
 ChainerCV supports functions commonly used to prepare data before feeding to a neural network.
```
Comment (Hakuyume) [inline README.md]: How about "ChainerCV provides ~"?

File: README.md [hunk 3]
```diff
@@ -63,7 +64,7 @@ img, label = dataset[0]
 
 As found in the example, `random_rotate` is one of the transforms ChainerCV supports. Like other transforms, this is just a
```
Comment (Hakuyume) [inline README.md]: How about "one of the transforms provided by ChainerCV"?

**Comment by Hakuyume** [comment]
From my understanding, `TransformDataset` will be removed from ChainerCV because it is merged into Chainer (Is it right?). So this documents should be changed after the release of Chainer v1.24.
Anyway, I added a few comments. Please check them.

**Comment by yuyu2172** [comment]
Thanks.
For `TransformDataset`, I will change the doc to recommend users to use the one merged to Chainer.
I am not sure about completely deleting it from `chainercv` namespace.

File: README.md (hunk 1 above)
Comment (yuyu2172) [inline README.md]: I think this is not a rst file, so `__getitem__` is fine.

**Comment by Hakuyume** [comment]
Why do you want to keep `TransformDataset` in ChainerCV? I think similar functions should be merged and  this is good time to remove it.

**Comment by yuyu2172** [comment]
It can be used as an appeal in the paper we may write.

File: README.md [hunk 4]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
 
 # Features
 
+## Models
+Currently, ChainerCV supports networks for object detection and image segmentation for 2D images.
+Image detection is the task of finding where objects are in an image, and classify the class to which the object belongs to.
```
Comment (Hakuyume) [inline README.md]: `finding objects in an image and classifying them` ?

File: README.md [hunk 5]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+Semantic segmentation is the task of segmentingimages into pieces and assigning object labels to them.
```
Comment (Hakuyume) [inline README.md]: `segmentingimages` -> `segmenting an image`

File: README.md [hunk 6]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+Our implementations include:
+
++ **Faster RCNN**
++ **Single Shot Multibox Detector (SSD)**
++ **SegNet**
```
Comment (Hakuyume) [inline README.md]: Is it better to show the task each model supports?
```
Detection
- Faster RCNN
- Single Shot Multibox Detector (SSD)
Semantic segmentation
- SegNet
```

File: README.md [hunk 7]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+
+The models support common interface across architectures among each task.
```
Comment (Hakuyume) [inline README.md]: How about `Models for a certain task are designed to have a common interface`?

File: README.md [hunk 8]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+For example, detection models support method that takes images and outputs coordinates, class labels and confidence scores of bounding boxes predicted around estimated regions of objects.
+The common interface allows users to swap different models easily inside their code.
+On top of that, the utilitiy codes build on top of this interface.
+For example, there is an extension `chainercv.extensions.detection_vis_report` that visualizes outputs during training for models that previously stated common interface.
```
Comment (Hakuyume) [inline README.md]: `extension` -> `extension of chainer.training.Trainer`
`chainercv.extensions.detection_vis_report` -> `chainercv.extensions.DetectionVisReport`

File: README.md [hunk 9]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+
+It is inconvenient to manually download pretrained models from the internet and passing file paths to Python objects to load them.
+ChainerCV alleviates this problem by internally downloading and storing files in a file system using Chainer's downloading mechanism.
```
Comment (Hakuyume) [inline README.md]: The order of sentences can be improved.

* models support to download pretrained weights automatically.
* it is inconvenient ~

File: README.md [hunk 10]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+```python
+from chainercv.links import FasterRCNNVGG16, SSD300
+model = FasterRCNNVGG16()  # or SSD300()
+# `bboxes` is a list of numpy arrays containing coordinates of boundnig boxes
+# around objects. `labels` and `confs` are class ids and confidence scores for
+# the boxes.
+bboxes, labels, confs = model.predict(imgs)  # imgs is a list of image
+```
+
+You can run a demo with a visualization with the commands below.
```
Comment (Hakuyume) [inline README.md]: `With the following command, you can run a visualization demo` ?

File: README.md [hunk 11]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+On top of that, the utilitiy codes build on top of this interface.
+For example, there is an extension `chainercv.extensions.detection_vis_report` that visualizes outputs during training for models that previously stated common interface.
+
+It is inconvenient to manually download pretrained models from the internet and passing file paths to Python objects to load them.
+ChainerCV alleviates this problem by internally downloading and storing files in a file system using Chainer's downloading mechanism.
+The convenient interface coupled with this functionality allows users to execute algorithms in two lines of code:
+
+```python
+from chainercv.links import FasterRCNNVGG16, SSD300
+model = FasterRCNNVGG16()  # or SSD300()
```
Comment (Hakuyume) [inline README.md]: Is it better to show `pretrained_model='voc07'` explicitly?

File: README.md [hunk 12]
```diff
@@ -35,14 +33,53 @@ Environments under Python 2.7.12 and 3.6.0 are tested.
...
+# the boxes.
+bboxes, labels, confs = model.predict(imgs)  # imgs is a list of image
+```
+
+You can run a demo with a visualization with the commands below.
+
+```
+cd examples/faster_rcnn  # or cd examples/ssd
+wget https://cloud.githubusercontent.com/assets/2062128/26187667/9cb236da-3bd5-11e7-8bcf-7dbd4302e2dc.jpg -O sample.jpg
+python demo.py -1 sample.jpg
```
Comment (Hakuyume) [inline README.md]: I guess `-1` means not using any GPUs. If so, it is better to make `gpu` argument optional. Please refer the demo code of SSD.

**Comment by Hakuyume** [comment]
LGTM
